src/test/c_api_tests.cpp

```diff
@@ -635,3 +650,160 @@ TEST_F(CapiInference, CallInferenceServerNotStarted) {
     OVMS_InferenceRequestDelete(request);
     OVMS_ServerDelete(cserver);
 }
+
+class CapiDagInference : public ::testing::Test {
+protected:
+    OVMS_ServerSettings* serverSettings = nullptr;
+    OVMS_ModelsSettings* modelsSettings = nullptr;
+    OVMS_Server* cserver = nullptr;
+    void SetUp() override {
+        std::string port = "9000";
+        randomizePort(port);
+        // prepare options
+        ASSERT_CAPI_STATUS_NULL(OVMS_ServerSettingsNew(&serverSettings));
+        ASSERT_NE(serverSettings, nullptr);
+        ASSERT_CAPI_STATUS_NULL(OVMS_ModelsSettingsNew(&modelsSettings));
+        ASSERT_NE(modelsSettings, nullptr);
+        ASSERT_CAPI_STATUS_NULL(OVMS_ServerNew(&cserver));
+        ASSERT_NE(cserver, nullptr);
+        ASSERT_CAPI_STATUS_NULL(OVMS_ServerSettingsSetGrpcPort(serverSettings, std::stoi(port)));
+    }
+    void TearDown() override {
+        OVMS_ServerDelete(cserver);
+        OVMS_ModelsSettingsDelete(modelsSettings);
+        OVMS_ServerSettingsDelete(serverSettings);
+        serverSettings = nullptr;
+        modelsSettings = nullptr;
+        cserver = nullptr;
+    }
+};
+
+TEST_F(CapiDagInference, BasicDummyDag) {
+    //////////////////////
+    // start server
+    //////////////////////
+    ASSERT_CAPI_STATUS_NULL(OVMS_ModelsSettingsSetConfigPath(modelsSettings, "/ovms/src/test/c_api/config_dummy_dag.json"));
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerStartFromConfigurationFile(cserver, serverSettings, modelsSettings));
+    ///////////////////////
+    // request creation
+    ///////////////////////
+    OVMS_InferenceRequest* request{nullptr};
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestNew(&request, cserver, "pipeline1Dummy", 1));
+    ASSERT_NE(nullptr, request);
+
+    // adding input
+    ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestAddInput(request, DUMMY_MODEL_INPUT_NAME, OVMS_DATATYPE_FP32, DUMMY_MODEL_SHAPE.data(), DUMMY_MODEL_SHAPE.size()));
+    // setting buffer
+    std::array<float, DUMMY_MODEL_INPUT_SIZE> data{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    uint32_t notUsedNum = 0;
```

Comment:
can be put to CapiDagInference fixture

Reply:
Both test have different inputs. What do you want to put in fixtue?

Reply:
`notUsedNum`, thats not a major issue, but could make the test shorter since it is required to pass to API calls